tests/unit/providers/__init__.py

```diff
@@ -1 +1,435 @@
 """Dependency injector providers unit tests."""
+import sys
```

Comment:
Not sure why did you move the tests here. Could we leave them on their place and just create a new module for contextvars singleton, e.g. `test_singletons_py3.py`?

Reply:
I think if I import `test_singletons_py2_py3.py` from `test_singletons_py3.py` the python2 tests will get run.

I could be wrong though.

I could move what I added to `__init__.py` to `singleton_test_common_utils.py` and import it from both files.